pkg/services/ngalert/schedule/schedule.go

```diff
@@ -299,6 +318,55 @@ func (sch *schedule) SyncAndApplyConfigFromDatabase() error {
 	return nil
 }
 
+func (sch *schedule) alertmanagersFromDatasources(orgID int64) ([]string, error) {
+	var alertmanagers []string
+	// We might have alertmanager datasources that are acting as external
+	// alertmanager, let's fetch them.
+	query := &datasources.GetDataSourcesByTypeQuery{
+		OrgId: orgID,
+		Type:  datasources.DS_ALERTMANAGER,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	err := sch.datasourceService.GetDataSourcesByType(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch datasources for org: %w", err)
+	}
+	for _, ds := range query.Result {
+		if !ds.JsonData.Get("handleGrafanaManagedAlerts").MustBool(false) {
```

Comment:
I think this is the same error as in the previous function?

```suggestion
		if ds.JsonData.Get("handleGrafanaManagedAlerts").MustBool(false) {
```

Reply:
No, if it's false you want to `continue`, if it's true generate the URL, this is correct.